src/NHibernate/Cfg/Settings.cs

```diff
@@ -18,6 +18,18 @@ namespace NHibernate.Cfg
 	/// </summary>
 	public sealed class Settings
 	{
+		private static ConfigurationProvider _configurationProvider = new StaticConfigurationManagerProvider();
+
+		/// <summary>
+		/// Provides ability to override default <see cref="System.Configuration.ConfigurationManager"/> with custom implementation.
+		/// Can be set to null if all configuration is specified by code 
+		/// </summary>
+		public static ConfigurationProvider ConfigurationProvider
```

Comment:
I do not like to have this property here. It seems only somehow related here.

However, I tried to put the property into Cfg.Environment or Cfg.SessionFactory classes as they seem to be more appropriate home for this property. Unfortunately, it did not work because these classes depend on logger and logger depends on the ConfigurationProvider. And so it creates a semi-circular dependency with undefined behaviour which resolves in TypeLoadException.

How about making this class fully independent by making a static property called `Current` in the `ConfigurationProvider` class itself? With the initial implementation (with interface) this was not really possible but without interface it is perfectly fine.

Reply:
Ok